New "Create a provider" page (#2012)
Qiskit/documentation: docs/open-source/create-a-provider.mdx

---
title: Create a provider
description: A short guide on integrating Qiskit into an external provider's quantum resources.     
---
# Integrate external quantum resources with Qiskit

> The Qiskit SDK is built to support third parties in creating external providers of quantum resources. This means that any organization which develops or deploys quantum compute resources can integrate their services into Qiskit and tap into its userbase. Doing so requires creating a package which supports requests for quantum compute resources and returns them to the user. Additionally, the package must allow users to submit jobs and retrieve their results, either through an implementation of the `qiskit.primitives` or `qiskit.provider` objects.

Reviewer: ```suggestion
The Qiskit SDK is built to support third parties in creating external providers of quantum resources.
This means that any organization which deploys resources that can be used to execute or simulate the execution of quantum program, it can be integrated  into Qiskit and tap into its userbase.
Doing so requires creating a package which supports requests a description of those quantum compute resources and list them to the user.
Additionally, the package must allow users to submit jobs and retrieve their results, either through an implementation of the `qiskit.primitives` objects.
```

Author: Splitting this up makes sense to do thanks. I edited the page to follow this and changed the wording of the last sentence just a bit.

---
title: Create a provider
description: A short guide on integrating Qiskit into an external provider's quantum resources.     
---
# Integrate external quantum resources with Qiskit

The Qiskit SDK is built to support third parties in creating external providers of quantum resources. 

This means that any organization which develops or deploys quantum compute resources can integrate their services into Qiskit and tap into its userbase. 

Doing so requires creating a package which supports requests for quantum compute resources and returns them to the user. 

Additionally, the package must allow users to submit jobs and retrieve their results through an implementation of the `qiskit.primitives` objects.

## Providing access to backends

> In order for users to transpile and execute `QuantumCircuit` objects using external resources, they need to instantiate an object containing a [`Target`](../api/qiskit/qiskit.transpiler.Target) which provides information about a QPU's constraints such as its connectivity, basis gates, and number of qubits. This can be provided through an interface similar to the [`QiskitRuntimeService`](../api/qiskit-ibm-runtime/qiskit_ibm_runtime.QiskitRuntimeService) through which a user can make requests for a QPU. This object should, at minimum, contain a `Target`, but a simpler approach would be to return an object which has inherited from either the [`BackendV2`](../api/qiskit/qiskit.providers.BackendV2) or [`BackendV1`](../api/qiskit/qiskit.providers.BackendV1) classes.

Reviewer: BackendV1 is deprecated and its usage is not recommended at this point https://github.com/Qiskit/qiskit/pull/12629/

```suggestion
In order for users to transpile and execute `QuantumCircuit` objects using external resources, they need to instantiate an object containing a [`Target`](../api/qiskit/qiskit.transpiler.Target) which provides information about a QPU's constraints such as its connectivity, basis gates, and number of qubits. This can be provided through an interface similar to the [`QiskitRuntimeService`](../api/qiskit-ibm-runtime/qiskit_ibm_runtime.QiskitRuntimeService) through which a user can make requests for a QPU. This object should, at minimum, contain a `Target`, but a simpler approach would be to return a [`BackendV2`](../api/qiskit/qiskit.providers.BackendV2) instance.
```

## Providing an interface for execution

> In addition to providing a service returning hardware configurations, a service providing access to external QPU resources also requires supporting the execution of quantum workloads. This can be done by creating implementations of the Qiskit primitives interfaces; for example the [`PubResult`](../api/qiskit/qiskit.primitives.PubResult), [`BasePrimitiveJob`](../api/qiskit/qiskit.primitives.BasePrimitiveJob), [`BaseEstimatorV2`](../api/qiskit/qiskit.primitives.BaseEstimatorV2), or [`BaseSamplerV2`](../api/qiskit/qiskit.primitives.BaseSamplerV2) among others. At minimum, these interfaces should be able to provide a method for execution, querying job status, returning the job results.      See the `qiskit.primitives` [API documentation](../api/qiskit/primitives) for more information.

Reviewer: ```suggestion
In addition to providing a service returning hardware configurations, a service providing access to external QPU resources also might also support execution of quantum workloads. Exposing that capability can be done by creating implementations of the Qiskit primitives interfaces; for example the (../api/qiskit/qiskit.primitives.BasePrimitiveJob), [`BaseEstimatorV2`](../api/qiskit/qiskit.primitives.BaseEstimatorV2) and [`BaseSamplerV2`](../api/qiskit/qiskit.primitives.BaseSamplerV2) among others. At minimum, these interfaces should be able to provide a method for execution, querying job status, and returning the job results.
See the `qiskit.primitives` [API documentation](../api/qiskit/primitives) for more information.
```
